File: python_programming_basics/nested_loops/e_03_sum_prime_non_prime.py
from math import sqrt


prime_sum = 0
non_prime_sum = 0

while True:
    command = input()

    if command == "stop":
        break

    current_number = int(command)

    if current_number < 0:
        print("Number is negative.")
        continue

    for number in range(2, int(sqrt(current_number)) + 1):
        if current_number % number == 0:
            non_prime_sum += current_number
            break

    else:
        prime_sum += current_number

print(f"Sum of all prime numbers is: {prime_sum}")
print(f"Sum of all non prime numbers is: {non_prime_sum}")

File: JavaScript Basics/06. Nested Loops/e-03-Sum-Prime-Non-Prime.js
function sumPrimesAndNonPrimes(input) {
    let sumPrimes = 0;
    let sumNonPrimes = 0;

    for (let value of input) {
        if (value === "stop") {
            break;
        }

        let number = Number(value);

        if (number < 0) {
            console.log("Number is negative.");
            continue;
        }

        if (number === 0 || number === 1) {
            sumNonPrimes += number;
            continue;
        }

        let isPrime = true;

        for (let i = 2; i <= Math.sqrt(number); i++) {
            if (number % i === 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            sumPrimes += number;
        } else {
            sumNonPrimes += number;
        }
    }

    console.log(`Sum of all prime numbers is: ${sumPrimes}`);
    console.log(`Sum of all non prime numbers is: ${sumNonPrimes}`);
}

sumPrimesAndNonPrimes(["3", "9", "0", "7", "19", "4", "stop"]);
sumPrimesAndNonPrimes(["30", "83", "33", "-1", "20", "stop"]);
sumPrimesAndNonPrimes(["0", "-9", "0", "stop"]);
